tests/layer_tests/pytorch_tests/test_sum.py

```diff
@@ -7,72 +7,97 @@
-
-    @pytest.mark.parametrize("axes,keep_dim",
-                             [(None, None), (None, False), (-1, None), (1, None), ((2, 3), False), ((3, 2), True)])
-    @pytest.mark.nightly
-    @pytest.mark.precommit
-    def test_sum(self, axes, keep_dim, ie_device, precision, ir_version):
-        self._test(*self.create_model(axes, keep_dim),
-                   ie_device, precision, ir_version)
-
-
-class TestSumBool(PytorchLayerTest):
-    def _prepare_input(self):
-        import numpy as np
-        return (np.random.randint(0, 2, (1, 3, 20, 24)).astype(bool),)
-
-    def create_model(self, axes, keep_dims):
-
-        import torch
-
-        class aten_sum(torch.nn.Module):
-            def __init__(self, axes=None, keep_dims=None):
-                super(aten_sum, self).__init__()
-                self.axes = axes
-                self.keep_dims = keep_dims
-
-            def forward(self, x):
-                x = x.to(torch.bool)
+                    if self.dtype is None:
+                        return torch.sum(x, self.axes)
+                    else:
+                        return torch.sum(x, self.axes, dtype=self.dtype)
+
+                if self.dtype is not None:    
+                    return torch.sum(x, self.axes, self.keep_dims, dtype=self.dtype)
+                else:
+                    return torch.sum(x, self.axes, self.keep_dims)
+
+            def forward_out(self, x, out):
+                x = x.to(self.input_dtype)
+                if self.axes is None and self.keep_dims is None:
+                     if self.dtype is None:
+                        return torch.sum(x, out=out), out
+                     else:
+                         return torch.sum(x, out=out, dtype=self.dtype), out
                 if self.axes is None and self.keep_dims is None:
-                    return torch.sum(x)
+                    if self.dtype is not None:
+                        return torch.sum(x, dtype=self.dtype, out=out), out
+                    else:
+                        return torch.sum(x, out=out), out
                 if self.axes is not None and self.keep_dims is None:
-                    return torch.sum(x, self.axes)
-                return torch.sum(x, self.axes, self.keep_dims)
+                    if self.dtype is not None:
+                        return torch.sum(x, self.axes, dtype=self.dtype, out=out), out
+                    else:
+                        return torch.sum(x, self.axes, out=out), out
+
+                if self.dtype is not None:
+                    return torch.sum(x, self.axes, self.keep_dims, dtype=self.dtype, out=out), out
+                else:
+                    return torch.sum(x, self.axes, self.keep_dims, out=out), out
 
         ref_net = None
 
-        return aten_sum(axes, keep_dims), ref_net, "aten::sum"
+        return aten_sum(input_torch_dtype, axes, keep_dims, torch_dtype, out), ref_net, "aten::sum"
 
     @pytest.mark.parametrize("axes,keep_dim",
```

Comment:
Overall, I see no issues :) My suggestion is to to rename:
```suggestion
    @pytest.mark.parametrize("axes,keep_dims",
```
To keep consistency with names used in the rest of the code.